front/components/assistant_builder/AssistantBuilderPreviewDrawer.tsx

```diff
@@ -372,3 +415,259 @@ const TemplateDropDownMenu = ({
+    if (agentConfigurationFeedbacks) {
+      // This triggers a re-fetch of the feedbacks.
+      setCurrentOldestFeedbackCreationDate(
+        new Date(
+          agentConfigurationFeedbacks[
+            agentConfigurationFeedbacks.length - 1
+          ].createdAt
+        )
+      );
+    }
+  }, [agentConfigurationFeedbacks]);
+
+  if (
+    isAgentConfigurationFeedbacksLoading ||
+    isAgentConfigurationHistoryLoading
+  ) {
+    return <Spinner />;
+  }
+
+  if (
+    !isAgentConfigurationFeedbacksLoading &&
+    (!feedbacks || feedbacks.length === 0)
+  ) {
+    return <div className="mt-3 text-sm text-element-900">No feedbacks.</div>;
+  }
+
+  if (!agentConfigurationHistory) {
+    return (
+      <div className="mt-3 text-sm text-element-900">
+        Error loading the previous agent versions.
+      </div>
+    );
+  }
+
+  return (
+    <div>
+      <div className="mb-2 flex flex-col gap-2">
+        <AgentConfigurationVersionHeader
+          agentConfiguration={agentConfigurationHistory[0]}
+          agentConfigurationVersion={agentConfigurationHistory[0].version}
+          isLatestVersion={true}
+        />
+        {feedbacks.map((feedback, index) => (
+          <div key={feedback.id} className="animate-fadeIn">
+            {index > 0 &&
+              feedback.agentConfigurationVersion !==
+                feedbacks[index - 1].agentConfigurationVersion && (
+                <AgentConfigurationVersionHeader
+                  agentConfiguration={agentConfigurationHistory?.find(
+                    (c) => c.version === feedback.agentConfigurationVersion
+                  )}
+                  agentConfigurationVersion={feedback.agentConfigurationVersion}
+                  isLatestVersion={false}
+                />
+              )}
+            <div className="mr-2">
+              <FeedbackCard
+                owner={owner}
+                feedback={feedback as AgentMessageFeedbackWithMetadataType}
+              />
+            </div>
+          </div>
+        ))}
+      </div>
+      {feedbacks && !feedbacksExhausted && (
+        <div className="mb-2 flex justify-center">
+          <Button
+            size="sm"
+            variant="outline"
+            label="Load more feedbacks"
+            onClick={handleLoadMoreFeedbacks}
+          />
+        </div>
+      )}
+    </div>
+  );
+};
+
+function AgentConfigurationVersionHeader({
+  agentConfigurationVersion,
+  agentConfiguration,
+  isLatestVersion,
+}: {
+  agentConfigurationVersion: number;
+  agentConfiguration: LightAgentConfigurationType | undefined;
+  isLatestVersion: boolean;
+}) {
+  const getAgentConfigurationVersionString = useCallback(
+    (config: LightAgentConfigurationType) => {
+      if (isLatestVersion) {
+        return "Latest version";
+      }
+      if (!config.versionCreatedAt) {
+        return `v${config.version}`;
+      }
+      return new Date(config.versionCreatedAt).toLocaleDateString("en-US", {
```

Comment:
We never display dates in the product so far. Let's stick to using timeAgoFrom with useLongFormat: true